pkg/controller/appinst_api.go

```diff
@@ -1050,6 +1064,25 @@ func (s *AppInstApi) createAppInstInternal(cctx *CallContext, in *edgeproto.AppI
 				if port.EndPort != 0 {
 					return fmt.Errorf("Shared IP access with port range not allowed")
 				}
+				if port.IsHTTP() {
```

Comment:
This is somewhat of a personal style question, but I guess I would say that I would really only split it out into a separate function if a) it needs to, because that code needs to be used somewhere else, or b) it is a large chunk of code that makes sense conceptually as a single unit that can be easily described by its function's name. Otherwise I tend to feel it's easier (at least for me) to understand the code if it's all in one place, and I don't need to be jumping around to various small little functions to understand exactly what they're doing. But I don't adhere to that religiously, just a general rule of thumb for me.